### PR TITLE
Cube.js Analytics Guide

### LINKED ISSUE #2271: How To Create Node Express Analytics Dashboard with Cube.js
We ask that you please be patient as our team works through approving and publishing all articles/tutorials in a timely manner. Allow 1-3 days for a topic to be reviewed and/or approved - allow 3-7 days for an articles to be reviewed and/or published.

Be sure to visit our [Resources Page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) for tools, resources, and example articles to go over.

**NOTE: (Please ensure that you have only one open issue + linked pull request at a time. This will ensure that we complete the article in a timely manner from inception to publishing.)**

**NOTE: We tend to stray away or tend not to publish reviews/comparisons of commercial product offerings.**

*We're looking for the first 2-3 paragraphs of the article that appropriately summarize what your article will be about.*

### Proposed title of article
How To Create Node Express Analytics Dashboard with Cube.js 

### Introduction paragraph (2-3 paragraphs):
 Cube.js solves a plethora of different problems every production-ready analytic application needs to solve: analytic SQL generation, query results caching and execution orchestration, data pre-aggregation, security, API for query results fetch, and visualization.
These features allow you to build production-grade analytics applications that are able to handle thousands of concurrent users and billions of data points. They also allow you to do analytics on a production MongoDB read replica or even a MongoDB main node due to their ability to reduce the amount of actual queries issued to a MongoDB instance.

In the following tutorial, I’ll show you how to create a basic analytics dashboard with Node, Express, and Cube.js. As a database, we’re going to use MongoDB with MongoDB BI Connector.

### Key takeaways: 
1. How to use MongoDB and BI connector.
2. Setting up a Backend with Express generator
3. Setting up and running Cube.js in serverless mode
4. Building Analytics Dashboard

### References: 
Please list links to any published content/research that you intend to use to support/guide this article.

### Templates to use as guides
- [How To Guide Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/how-to-guide-template.md)
- [Software Review Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/software-review-template.md)
- [Tutorial Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/tutorial-template.md)

### PR DESCRIPTION
**Pull request by bhanjibrilliant**
### Article checklist

**ATTENTION: In your PR - add the ISSUE (topic form) # that this PR corresponds to. You can do this by also adding a comment that states **["This fixes #(enter your Idea Content Form #)"](https://youtu.be/TKJ4RdhyB5Y?t=278)** to link and close your Idea Content (issue) when the PR is merged.**
      
      
**NOTE: (Please ensure that you have only one open issue + linked pull request at a time. This will ensure that we complete the article in a timely manner from inception to publishing.)**

**If you have not already please go over our [Resources Page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) for more tips and tricks on how to improve our overall technical writing.**

#### Preliminary Checks - Formatting and Structure 
- [x] Is the article you are submitting an in-depth and unique article? Does it go beyond what is in the official docs and what is covered in other blog sites. See these articles as [examples](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md).
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**
- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**
- [x] Have you included a hero image that is 600x400 pixels and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
- [x] In your PR - add a comment that states **["This fixes #(enter your Idea Content Form #)"](https://youtu.be/TKJ4RdhyB5Y?t=278)** to link and close your Idea Content (issue) when the PR is merged.
      
#### Grammar & plagiarism checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) -  [Grammarly](https://grammarly.com) 
- [x] Please place your article through a 3rd party plagiarism checker? We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [x] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**

### Contribution guidelines
For first-time contributors and for more details, **see [Contributing Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**
This fixes #2271

### CODE REVIEW

**Comment by hectorkambow** [comment]
@bhanjibrilliant A few things with this PR . 

One - please always ask to have the issue re-opened before submitting the PR. 

Two - this PR was approved a very long time ago - and since then we have refrained but approving/publishing anymore Node.js related articles - as we have many in our EngEd library. 

Three - this tutorial seems rather light on substance. 

Had I seen this topic being re-opened - I would have mentioned that we are no longer publishing articles in this topic area.

**Comment by hectorkambow** [comment]
plagiarism check done

File: content/articles/mongo-nodejs-cubejs-dashboard/index.md [hunk 1]
```diff
@@ -0,0 +1,249 @@
...
+- [Objectives](#objectives)
+- [Setting up the development environment](#setting-up-the-development-environment)
+- [Generating Cube.js schema](#generating-cubejs-schema)
+- [Adding Cube.js environmental variables](#adding-cubejs-environmental-variables)
+- [Conclusion](#conclusion)
+
+### Prerequisites
+
+- This tutorial requires a basic understanding of the MongoDB and MongoDB BI connector.
+- Basic knowledge in Node.js, Express, and Cube.js.
```
Comment (jumamiller) [inline content/articles/mongo-nodejs-cubejs-dashboard/index.md:19]: @bhanjibrilliant, could give links to cube.js. Or rather explain what it does. Do not let readers scratch their heads over something you could as well explain. I found a few instances of this, kindly fix them

File: content/articles/mongo-nodejs-cubejs-dashboard/index.md [hunk 2]
```diff
@@ -0,0 +1,249 @@
...
+     $ npm install
+
+   run the app:
+     $ DEBUG=example-analytics-dashboard:* npm start
+
+```
+
+**Step 3:**
+Let's now install the local MongoDB instance and MongoDB BI connector.  
+Begin by running the following command to import the MongoDB instance public key:
```
Comment (jumamiller) [inline content/articles/mongo-nodejs-cubejs-dashboard/index.md:91]: @bhanjibrilliant, imagine of a reader `hearing of the term 'BI connector'` for the first time, what would they do? make your article flow as much as possible. Do not let your readers leave your article. Kindly make this change.

**Review by jumamiller — CHANGES_REQUESTED** [review]
Hi @bhanjibrilliant I've suggested a few changes, kindly work on them before we proceed to next step. Meanwhile tech review is done. Great work!

**Review by jumamiller — APPROVED** [review]
@bhanjibrilliant, I've made some edits on your article. Kindly ensure you over and check to improve the quality of your articles in the future. Also, keep in mind the statement @hectorkambow made previously. Otherwise it's a great and easy article to follow! Kudos!

**Comment by LinusMuema** [comment]
Hi @bhanjibrilliant 
Before I even continue with the review, I noticed your hero image is square in shape but section requires a 600 x 400 image. Please rectify this and ping me so I can continue with the review. 

@jumamiller, please make sure you check this in further articles.

**Comment by bhanjibrilliant** [comment]
@LinusMuema, I've updated the hero image

**Comment by LinusMuema** [comment]
Hi @bhanjibrilliant 
Your author details are missing. Could you please check on that so I can merge this article

**Comment by bhanjibrilliant** [comment]
> Hi @bhanjibrilliant
> Your author details are missing. Could you please check on that so I can merge this article

I had provided them before, lemme confirm

**Comment by bhanjibrilliant** [comment]
https://github.com/section-engineering-education/engineering-education/tree/master/content/authors/bhanji-brilliant

**Comment by LinusMuema** [comment]
It's not available in your pr. |
I will merge the article and we'll check if it is available in the master branch. But in the future please ensure it shows up in your repo and fork.

**Comment by LinusMuema** [comment]
Hi @bhanjibrilliant 
The article is great and keep up the great work. Just check my commits for changes in flow and grammar.
Give it a few mins and check the author details in your article

cc: @jumamiller

**Pull request merged**